source/dem/lagrangian_post_processing.cc

```diff
@@ -173,20 +173,31 @@ LagrangianPostProcessing<dim>::calculate_cell_average_particles_velocity(
 
 template <int dim>
 void
-LagrangianPostProcessing<dim>::write_average_particles_velocity(
+LagrangianPostProcessing<dim>::write_post_processing_results(
   const parallel::distributed::Triangulation<dim> &triangulation,
   PVDHandler &                                     grid_pvdhandler,
+  const Particles::ParticleHandler<dim> &          particle_handler,
   const DEMSolverParameters<dim> &                 dem_parameters,
-  const double                                     time,
+  DoFHandler<dim> &                                background_dh,
+  const double                                     current_time,
   const unsigned int                               step_number,
   const MPI_Comm &                                 mpi_communicator)
 {
   const std::string folder = dem_parameters.simulation_control.output_folder;
+  const std::string particles_solution_name =
+    dem_parameters.simulation_control.output_name;
+  const unsigned int group_files =
+    dem_parameters.simulation_control.group_files;
 
   DataOut<dim> data_out;
   data_out.attach_triangulation(triangulation);
 
-  std::vector<std::string> average_solution_names(1, "average_velocity_x");
+  std::vector<std::string> average_solution_names;
+
+  // Write particles' average velocity
+  calculate_average_particles_velocity(triangulation, particle_handler);
+
+  average_solution_names.push_back("average_velocity_x");
   average_solution_names.push_back("average_velocity_y");
   if constexpr (dim == 3)
     average_solution_names.push_back("average_velocity_z");
```

Comment:
I was thinking about the same thing

Reply:
@voferreira I think storage is [avg_x_0, avg_y_0, avg_z_0, avg_x_1, ...] You can look at line 1570 of `navier_stokes_base.cc`, but not 100% sure since it's DoF data and not cell data, but it's should be similar. :)